### ISSUE TITLE
 Global Search (Feature Request)

### ISSUE THREAD

**Issue by 1ubuntuuser** [frappe/crm#439]
Erpnext has the Awesome Bar. Very handy for finding info. 

If you're on the phone with a customer and don't know if they are a lead, a deal, or just a contact, being able to search their name, phone, or email to pull up list entries would be very useful.

Can we subclass the existing Awesome Bar somehow? Of course, it would have to be filtered only to include doctypes in the FCRM module.

Is this something that is being considered?

**Comment by 1ubuntuuser** [comment]
@shariquerik I know you're busy doing stuff, but we could be multiplying your efforts with some guidance. I have been burnt in the past developing features that where not wanted, so never got the PR over the line. Is this a feature that is wanted?

**Comment by shariquerik** [comment]
@1ubuntuuser Yes it is a required feature. Not only the doctypes in FCRM module but also Contact & Email Template doctype which is not part of FCRM module

**Comment by shariquerik** [comment]
If you are planning to work on it we do have a design
<img width="896" alt="image" src="https://github.com/user-attachments/assets/f8197d9d-629e-4c3b-a1d5-82827607a536">
<img width="878" alt="image" src="https://github.com/user-attachments/assets/dd2f2549-0267-45dd-b842-e42f222bf30e">
<img width="870" alt="image" src="https://github.com/user-attachments/assets/e238e6b4-e9ce-4e69-8af1-38eae52bbde9">
<img width="890" alt="image" src="https://github.com/user-attachments/assets/79d7940c-8827-43c2-a775-5531c5d0ddc7">

**Comment by 1ubuntuuser** [comment]
@shariquerik Looks very functional. I'll keep chipping away at the email side for now. That looks more involved than just porting the functionality from ERPNext.


Is there a developer channel, or some sort of roadmap for features?

**Comment by ivangretsky** [comment]
Thank you, friends! I just wanted to let you know that this feature is crucial and desired indeed! So please be sure about that. Hope this can be implemented soon, as for now it seems to be not possible to even search across all fields of a certain type, not even across types.

**Comment by Facecloth4523** [comment]
Any update on a global search function? My company refuses to consider Frappe CRM until this function is implemented

**Comment by girish3** [comment]
@shariquerik @shahzeelahmed @michellealva Any plans for global search? We have too many leads/deals on our system, and our team needs a way to quickly jump to a particular lead/deal through name/organisation search. Full-text search will be preferred. Thank you.

**Comment by trufurs** [comment]
we did some js injection to add search bar with search functionality with [rtCamp/frappe_crm_xt](https://github.com/rtCamp/frappe_crm_xt/wiki#feature-tour)
to search for leads and deals. Using [rtCamp/frappe_search ](https://github.com/rtCamp/frappe_search)
